WordPress/gutenberg
# [Widgets Editor] Add a filter function to `registerCoreBlock`

## Description

Fix #26178.

Add a `filter` function to `registerCoreBlock` and filter out `core/more` block in widgets editor.

I'm not sure this is the right approach. Another option I can think of is to add a `disallowedBlocks` to `<InnerBlocks>` and `widget-area`. Or remove `more` block from core, and register them manually one-by-one in each usage. I figured to open this PR first to gain more discussion and suggestions.

## How has this been tested?



1. Go to Widgets editor
2. Insert a new block
3. Search for the "more" block
4. Observe that there's no "more" block

## Screenshots 
![image](https://user-images.githubusercontent.com/7753001/96418358-3cea2580-1225-11eb-9ca0-18c2fa7510e3.png)

## Types of changes




Bug fix

## Checklist:
